osutil/statelock.go

```diff
@@ -0,0 +1,131 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//go:build statelock
+
+/*
+ * Copyright (C) 2021 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package osutil
+
+import (
+	"fmt"
+	"os"
+	"runtime"
+	"time"
+)
+
+var (
+	traceStateLock = false
+
+	traceThreshold = int64(0)
+	traceFilePath  = ""
+)
+
+func init() {
+	err := func() error {
+		if !GetenvBool("SNAPPY_TESTING") {
+			return fmt.Errorf("SNAPPY_TESTING not set")
+		}
+
+		threshold := GetenvInt64("SNAPD_STATE_LOCK_TRACE_THRESHOLD_MS")
+		logFilePath := os.Getenv("SNAPD_STATE_LOCK_TRACE_FILE")
+
+		if threshold <= 0 {
+			return fmt.Errorf("SNAPD_STATE_LOCK_TRACE_TREHSHOLD_MS is unset, invalid or 0")
+		}
+
+		if logFilePath == "" {
+			return fmt.Errorf("SNAPD_STATE_LOCK_TRACE_FILE is unset")
+		}
+
+		traceThreshold = threshold
+		traceFilePath = logFilePath
+
+		return nil
+	}()
+
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "cannot enable state lock tracing: %v\n", err)
```

Comment:
I'm not sure it make sense to have all the testing snapds print this?

Reply:
dropped